language/compiler/ir-to-bytecode/syntax/src/syntax.rs

```diff
@@ -240,34 +240,26 @@ fn parse_copyable_val_<'input>(
     Ok(spanned(start_loc, end_loc, val))
 }
 
-// Exp = BinopExp = CmpOpExp
-// CmpOpExp = (OrExp CmpOp OrExp) || OrExp
-// OrExp = (AndExp OrOp AndExp) || AndExp
-// AndExp = (XorExp AndOp XorExp) || XorExp
-// XorExp = (BinOrExp XorOp BinOrExp) || BinOrExp
-// BinOrExp = (BinAndExp BinOrOp BinAndExp) || BinAndExp
-// BinAndExp = (AddSubExp BinAndOp AddSubExp) || AddSubExp
-// AddSubExp = (FactorExp AddSubOp FactorExp) || FactorExp
-// FactorExp = (UnaryExp FactorOp UnaryExp) || UnaryExp
-
+// Get the precedence of a binary operator.
 fn get_precedence(token: &Tok) -> u32 {
     match token {
-        Tok::EqualEqual => 1,
-        Tok::ExclaimEqual => 1,
-        Tok::Less => 1,
-        Tok::Greater => 1,
-        Tok::LessEqual => 1,
-        Tok::GreaterEqual => 1,
+        // Reserved minimum precedence value is 1 (specified in parse_exp_)
```

Comment:
I'm a bit confused here. Why are we reserving 1? instead of making `Tok::PipePipe => 1`?

Reply:
The initial call to parse_rhs_of_binary_exp() in parse_exp_ needs to specify the minimum precedence value. To avoid hardcoding a specific value, I had previously been using the precedence value for what I knew (at the time) to be the lowest precedence operator. However, with this change, what was previously the lowest precedence ("==") is no longer that. Using a specific value ("1") seems like it more directly captures the intention of the code without relying on any particular ordering of the operators. But, if the code relies on the minimum precedence having the value of 1, then we should make that assumption clear at the point where the precedence values are defined. There's no need to reserve the value, but I thought that would help draw attention to it. For example, if we change get_precedence() to return an Option type, then it would be a problem if someone decides to renumber the precedence values to begin at zero without adjusting the call to parse_rhs_of_binary_exp.